### PR TITLE
Change the way how tests renders components

### PR DESCRIPTION
**Pull request by DawidHaratyk**
I've also fixed tests about leader buttons visibility

### CODE REVIEW

File: team-fingerprints-web/src/routes/TeamManagement/EmailWhitelist/__tests__/data.ts [hunk 1]
```diff
@@ -73,7 +73,7 @@ export const superAdminRoles: CompanyRole[] = [
     teamId: "100",
     createdAt: "12:45",
     email: "koko@gmail.com",
-    role: "SUPER_ADMIN",
```
Comment (tobob) [inline team-fingerprints-web/src/routes/TeamManagement/EmailWhitelist/__tests__/data.ts:76]: @DawidHaratyk  don't you want to use factories to create that structure?

File: team-fingerprints-web/src/routes/TeamManagement/EmailWhitelist/__tests__/EmailWhitelist.spec.tsx [hunk 2]
```diff
@@ -3,31 +3,37 @@ import { render, screen, within } from "@testing-library/react";
...
+import { AppSetup } from "/testing/helpers/AppSetup";
 
 const mockedMakeALeader = jest.fn();
 
 jest.mock("@auth0/auth0-react", () => ({
   ...jest.requireActual("@auth0/auth0-react"),
   useAuth0: () => ({
-    user: auth0User,
+    user: {
+      email: "someone@gmail.com",
```
Comment (tobob) [inline team-fingerprints-web/src/routes/TeamManagement/EmailWhitelist/__tests__/EmailWhitelist.spec.tsx:16]: maybe using faker will be better there, and also use faker to mock this @onet.pl email

**Review by tobob — CHANGES_REQUESTED** [review]

**Comment by tobob** [comment]
@shirve please also take a look at this and any feedback means a lot

**Comment by shirve** [comment]
@DawidHaratyk I refactored some of your tests in #564 so please take a look there.

- There is no need to mock auth0 as we already have that mocked in `setupTests`
- It would be preffered to use factories rather than static mocks
- If you are rendering the same components structure in each test it would be preferred to take that render method outside to some function like `renderComponent` to make it more readable and to keep the code DRY

**Comment by tobob** [comment]
@shirve so we have a problem because you both guys were refactoring the same files/code

please try to avoid such a situation in the future, because now we need to resolve massive merge conflicts

**Comment by shirve** [comment]
@tobob 
these tests have been merged in the morning today in #551 that's why I thought Dawid was not going to work on them
tut that's my bad for not asking tho

**Pull request closed without merging**